components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger_test.go

```diff
@@ -0,0 +1,272 @@
+			},
+			{
+			  "title": "Sudo password invalid",
+			  "passed": false,
+			  "success_msg": "",
+			  "error_msg": "SSH user sudo password is invalid for the node with IP <node_ip>",
+			  "resolution_msg": "Ensure you have provided the correct sudo password and the user has sudo access on the node: <node_ip>"
+			}
+		  ]
+		}
+	  }`
+)
+
+func GetRequestJson() models.Config {
+	ipConfig := models.Config{}
+
+	json.Unmarshal([]byte(`{
+		  "ssh_user": {
+			"user_name": "ubuntu",
+			"private_key": "test_key",
+			"sudo_password": "test@123"
+		  },
+		  "arch": "existing_nodes",
+		  "backup": {
+			"file_system": {
+			  "mount_location": "/mnt/automate_backups"
+			}
+		  },
+		  "hardware": {
+			"automate_node_count": 1,
+			"automate_node_ips": [
+			  "1.2.3.4"
+			],
+			"chef_infra_server_node_count": 1,
+			"chef_infra_server_node_ips": [
+			  "5.6.7.8"
+			],
+			"postgresql_node_count": 1,
+			"postgresql_node_ips": [
+			  "9.10.11.12"
+			],
+			"opensearch_node_count": 1,
+			"opensearch_node_ips": [
+			  "14.15.16.17"
+			]
+		  },
+		  "certificate": {
+			"fqdn":"my_fqdn",
+			"root_cert":"---- VALID ROOT CA ----",
+			"nodes":[
+				{
+					"ip":"1.2.3.4",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"5.6.7.8",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"9.10.11.12",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				},
+				{
+					"ip":"14.15.16.17",
+					"cert":"---- VALID NODE CERT ----",
+					"key":"---- VALID PRIVATE KEY ----",
+					"admin_key":"---- VALID ADMIN PRIVATE KEY ----",
+					"admin_cert":"---- VALID ADMIN CERT ----"
+				}
+			]
+		  }
+		}`), &ipConfig)
+	return ipConfig
+}
+
+func startMockServerOnCustomPort(mockServer *httptest.Server, port string) error {
+	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%s", port))
+	if err != nil {
+		return err
+	}
+	mockServer.Listener = l
+	mockServer.Start()
+	return nil
+}
+
+func TestNewSshUserAccessCheck(t *testing.T) {
+	testPort := "1234"
+	cc := NewSshUserAccessCheck(logger.NewTestLogger(), testPort)
+	assert.NotNil(t, cc)
+	assert.NotNil(t, cc.log)
+	assert.Equal(t, constants.LOCAL_HOST_URL, cc.host)
+	assert.Equal(t, testPort, cc.port)
+}
+
+func TestSshUserAccessCheck_Run(t *testing.T) {
+	t.Run("Returns OK", func(t *testing.T) {
+		//starting the mock server on custom port
+		mockServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
you should assert the input body that you are posting, that should be similar to API request body

Reply:
As there will be multiple different IPs, i have created a different test to validate the request.

Reply:
Added after the discussion on the call